src/main/java/sysc4806/graduateAdmissions/model/Privilege.java

```diff
@@ -0,0 +1,39 @@
+package sysc4806.graduateAdmissions.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * This class is used to specify privileges for roles. A Role with have a collection
+ * of these objects to determine what it can and cannot do. A Privilege is made up of three
+ * parts: An action (eg. create, delete), a target for that action (eg. application, interest),
+ * and an owner of that target (eg. self, everyone). Combinations of these allow the
+ * specification of many different privileges for roles.
+ *
+ * For example, a student would have privileges for creating and deleting their own applications.
+ * Admin staff would have privileges for creating and deleting any student's application.
+ *
+ * @author luke
+ */
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class Privilege {
```

Comment:
I'd recommend adding the @ Builder annotation. It makes object creation pretty nice. I'd leave it up to you, but this will set a precedent for all other objects we create that are like this. It just adds some neat functionality. Check it out if you havent already: https://projectlombok.org/features/Builder

Reply:
This will probably be a common type of comment since many of us have not used these libraries before. I can definitely see the appeal of this and I will replace the constructor annotations with this